mypy/fastparse.py

```diff
@@ -647,13 +652,25 @@ def visit_Num(self, n: ast35.Num) -> Node:
     # Str(string s) -- need to specify raw, unicode, etc?
     @with_line
     def visit_Str(self, n: ast35.Str) -> Node:
-        return StrExpr(n.s)
+        if self.pyversion[0] >= 3 or self.is_stub:
+            # Hack: assume all string literals in Python 2 stubs are normal
+            # strs (i.e. not unicode).  All stubs are parsed with the Python 3
+            # parser, which causes unprefixed string literals to be interpreted
+            # as unicode instead of bytes.  This hack is generally okay,
+            # because mypy considers str literals to be compatible with
+            # unicode.
+            return StrExpr(n.s)
+        else:
+            return UnicodeExpr(n.s)
 
     # Bytes(bytes s)
     @with_line
     def visit_Bytes(self, n: ast35.Bytes) -> Node:
         # TODO: this is kind of hacky
```

Comment:
Do you still consider this hacky? I propose to remove the comments from the bodies of visit_Str() and visit_Bytes() and instead adding a block comment before the former explaining what's going on with both. Also remove the confusing remark "need to specify raw, unicode, etc?" from the former.

I also propose to update mypy/nodes.py with a block comment before class StrExpr explaining what each is used for, e.g. like this:

- Python 2
  b'x', 'x' -> StrExpr
  u'x' -> UnicodeExpr

- Python 3
  b'x' -> BytesExpr
  'x', u'x' -> StrExpr

(assuming I got it right).

It wouldn't hurt repeating all that information in typed_ast/conversions.py just before visit_Str(), plus the info how ast27 maps literals to nodes:

  b'x', 'x' -> Str(s=b'x')
  u'x' -> Str(s=u'x')

I know the u prefix is redundant in Python 3, but it's still a helpful reminder